sklearn/preprocessing/tests/test_encoders.py

```diff
@@ -398,6 +398,15 @@ def test_one_hot_encoder_feature_names_drop(drop, expected_names):
     assert_array_equal(expected_names, feature_names)
 
 
+def test_one_hot_encoder_drop_equals_if_binary():
+    X = [['Male', 1], ['Female', 3], ['Female', 2]]
```

Comment:
You can add three additional columns (one binary and one mutli-category and again one binary).

We can ensure that there is nothing tricky with the ordering

Reply:
Made all the changes